IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m

```diff
@@ -0,0 +1,131 @@
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <AuthenticationServices/AuthenticationServices.h>
+#import "ASAuthorizationSingleSignOnProvider+MSIDExtensions.h"
+#import "MSIDSSOExtensionInteractiveTokenRequest.h"
+#import "MSIDInteractiveTokenRequest+Internal.h"
+#import "MSIDJsonSerializer.h"
+#import "MSIDInteractiveRequestParameters.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDAuthority.h"
+#import "MSIDSSOExtensionTokenRequestDelegate.h"
+#import "MSIDBrokerOperationInteractiveTokenRequest+InteractiveParameters.h"
+#import "MSIDBrokerOperationInteractiveTokenRequest+SSORequest.h"
+
+@interface MSIDSSOExtensionInteractiveTokenRequest () <ASAuthorizationControllerPresentationContextProviding>
+
+@property (nonatomic) ASAuthorizationController *authorizationController;
+@property (nonatomic, copy) MSIDInteractiveRequestCompletionBlock requestCompletionBlock;
+@property (nonatomic) MSIDSSOExtensionTokenRequestDelegate *extensionDelegate;
+@property (nonatomic) ASAuthorizationSingleSignOnProvider *ssoProvider;
+
+@end
+
+@implementation MSIDSSOExtensionInteractiveTokenRequest
+
+- (instancetype)initWithRequestParameters:(MSIDInteractiveRequestParameters *)parameters
+                             oauthFactory:(MSIDOauth2Factory *)oauthFactory
+                   tokenResponseValidator:(MSIDTokenResponseValidator *)tokenResponseValidator
+                               tokenCache:(id<MSIDCacheAccessor>)tokenCache
+                     accountMetadataCache:(MSIDAccountMetadataCacheAccessor *)accountMetadataCache
+{
+    self = [super initWithRequestParameters:parameters
+                               oauthFactory:oauthFactory
+                     tokenResponseValidator:tokenResponseValidator
+                                 tokenCache:tokenCache
+                       accountMetadataCache:accountMetadataCache];
+
+    if (self)
+    {
+        _extensionDelegate = [MSIDSSOExtensionTokenRequestDelegate new];
+        _extensionDelegate.context = parameters;
```

Comment:
nit: If I remember correctly, we generally don't store context, but pass it around. Not much difference, just consistency question.

Reply:
In this case, it needs to be property because we are using it for logging in apple's callbacks, like here:

`- (void)authorizationController:(ASAuthorizationController *)controller didCompleteWithAuthorization:(ASAuthorization *)authorization`